runtime/fail_byt.c

```diff
@@ -39,9 +39,10 @@ CAMLexport void caml_raise(value v)
   caml_channel_cleanup_on_raise();
 
   // avoid calling caml_raise recursively
-  v = caml_process_pending_actions_with_root_exn(v);
-  if (Is_exception_result(v))
-    v = Extract_exception(v);
+  {
+    caml_result result = caml_process_pending_actions_with_root_result(v);
+    v = result.data;
+  }
```

Comment:
The blanket `v = result.data;` is very unusual so it could benefit from a comment saying that this is really intended.

It looks like one can do without the braces here.

Reply:
New version.

```c
  caml_result result = caml_process_pending_actions_with_root_result(v);
  /* If the result is a value, we want to assign it to [v].
     If the result is an exception, we want to continue raising it instead of [v].
     The line below does both these things at once. */
  v = result.data;
```